File: domoclient/src/main/java/net/glenmazza/domoclient/service/AbstractRESTService.java
package net.glenmazza.domoclient.service;

import com.fasterxml.jackson.databind.ObjectMapper;
import net.glenmazza.domoclient.util.JSONUtils;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.web.reactive.function.client.WebClient;

public abstract class AbstractRESTService {

    private static final Logger LOGGER = LoggerFactory.getLogger(AbstractRESTService.class);

    @Value("${domo.api.base-url}")
    protected String baseUrl;

    protected WebClient webClient;

    protected final ObjectMapper objectMapper;

    public AbstractRESTService() {
        this.objectMapper = JSONUtils.createObjectMapper();
    }

    protected AbstractRESTService(WebClient webClient) {
        this();
        this.webClient = webClient;
    }

    public void setWebClient(WebClient webClient) {
        this.webClient = webClient;
    }

}

File: splashclient/src/main/kotlin/net/glenmazza/splashclient/service/AbstractRESTService.kt
package net.glenmazza.splashclient.service

import com.fasterxml.jackson.databind.ObjectMapper
import net.glenmazza.splashclient.util.createObjectMapper
import org.springframework.beans.factory.annotation.Value
import org.springframework.web.reactive.function.client.WebClient

abstract class AbstractRESTService(val webClient: WebClient) {

    @Value("\${splash.api.base-url}")
    protected var baseUrl: String? = null

    protected val objectMapper: ObjectMapper? = createObjectMapper()
}
